### PR TITLE
Add convenience properties for the DecodeError.

### PR DESCRIPTION
**Pull request by jshier**
This PR adds `error` and `errorString` properties to `Decoded`, making it more convenient to extract `DecodeError`s and their `description`s.

Would you like any tests?

### CODE REVIEW

File: Argo/Types/Decoded.swift [hunk 1]
```diff
@@ -8,6 +8,17 @@ public enum Decoded<T> {
...
   }
+  
+  public var error: DecodeError? {
+    switch self {
+    case let .Failure(error): return error
+    default: return .None
+    }
+  }
+    
+  public var errorString: String? {
```
Comment (tonyd256) [inline Argo/Types/Decoded.swift]: not sure this one is worth it ... thoughts @gfontenot?

**Comment by tonyd256** [comment]
I'm definitely ok with the `error` addition. Still unsure about `errorString`.

File: Argo/Types/Decoded.swift (hunk 1 above)
Comment (gfontenot) [inline Argo/Types/Decoded.swift]: Agreed. I don't think we need this.

**Comment by jshier** [comment]
@tonyd256 @gfontenot I've removed errorString.

**Comment by tonyd256** [comment]
@jshier Great thanks! If you could add a test for this as well, that would be awesome!

**Comment by tonyd256** [comment]
@jshier I'm thinking about cutting a new release tomorrow. Would you be able to get this in?

**Comment by jshier** [comment]
I'll try to get it done by noon EST.

**Comment by jshier** [comment]
@tonyd256 I've added tests, let me know if you think they're appropriate.

**Comment by jshier** [comment]
@tonyd256 Do you want any mention of this property in the documentation? I know it's still in flux, so I haven't touched anything.

**Comment by tonyd256** [comment]
This is great! Thanks!

**Comment by tonyd256** [comment]
Merged here: b2ebd2ff8a2de71e34a1d0b16af30a28bef6158b

**Pull request closed without merging**